x-pack/plugins/observability_ai_assistant/public/plugin.tsx

```diff
@@ -117,9 +120,50 @@ export class ObservabilityAIAssistantPlugin
       });
     });
 
+    const withProviders = <P extends {}, R = {}>(
+      Component: ComponentType<P>,
+      services: Omit<CoreStart, 'plugins'> & {
+        plugins: { start: ObservabilityAIAssistantPluginStartDependencies };
+      }
+    ) =>
+      React.forwardRef((props: P, ref: Ref<R>) => (
+        <KibanaContextProvider services={services}>
+          <ObservabilityAIAssistantProvider value={service}>
```

Comment:
Should the provider not be lazily loaded?

Reply:
It's more about some import in the provider inadvertently sneaking in and bloating the bundle size.

Reply:
Ok, will add lazy loading there as well